interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp

```diff
@@ -869,6 +865,11 @@ namespace cling {
     return ""; // Search found no match.
   }
 
+  DynamicLibraryManager::~DynamicLibraryManager() {
+    static_assert(sizeof(Dyld) > 0, "Incomplete type");
```

Comment:
The next line is ill-formed without definition of `m-Dyld`'s type. What does this `static_assert`add?

Reply:
The static asserts enforces the compiler to know the size. Even if the next line was ill-formed, it compiled with incomplete type without even a warning. That caused the destructor for Dyld to not be called, leading to significant memory leaks.

Reply:
Ill-formed means it doesn't compile. And I'm wrong: https://www.godbolt.org/z/1ovE34